File: content-solutions/implementation/sim-dice-sum.js
function solution(commands) {
  let [top, bottom, north, south, east, west] = [1, 6, 2, 5, 3, 4];
  let total = 0;
  for (const ch of commands) {
    if (ch === "E") [top, east, bottom, west] = [west, top, east, bottom];
    else if (ch === "W") [top, west, bottom, east] = [east, top, west, bottom];
    else if (ch === "N") [top, north, bottom, south] = [south, top, north, bottom];
    else [top, south, bottom, north] = [north, top, south, bottom];
    total += top;
  }
  return total;
}

File: content-solutions/implementation/sim-dice-sum.java
import java.util.*;

class Solution {
    public int solution(String commands) {
        int top = 1, bottom = 6, north = 2, south = 5, east = 3, west = 4;
        int total = 0;
        for (char ch : commands.toCharArray()) {
            int t = top;
            if (ch == 'E') {
                top = west;
                west = bottom;
                bottom = east;
                east = t;
            } else if (ch == 'W') {
                top = east;
                east = bottom;
                bottom = west;
                west = t;
            } else if (ch == 'N') {
                top = south;
                south = bottom;
                bottom = north;
                north = t;
            } else {
                top = north;
                north = bottom;
                bottom = south;
                south = t;
            }
            total += top;
        }
        return total;
    }
}
